src/utils/platform.ts

```diff
@@ -14,12 +15,42 @@ export const getPlatformShortName = (
   return 'Unspecified';
 };
 
+const osMapping = {
+  linux64: 'Linux x64',
+  linux1804: 'Linux 18.04',
+  linux2204: 'Linux 22.04',
+  osx: 'OS X',
+  macosx1014: 'OS X 10.14',
+  macosx1015: 'OS X 10.15',
+  macosx1100: 'OS X 11',
+  macosx1300: 'macOS 13',
+  macosx1400: 'macOS 14.00',
+  macosx1470: 'macOS 14.70',
+  win32: 'Windows x86',
+  win64: 'Windows x64',
+  windows7: 'Windows 7',
+  windows10: 'Windows 10',
+  windows11: 'Windows 11',
+  windows2012: 'Windows 2012',
+  windows: 'Windows',
+};
+
+const extractPlatformWithOs = (platform: string) => {
+  const name = osMapping[platform.split('-')[0] as PlatformOS];
+
+  if (name) return name;
+
+  return getPlatformShortName(platform);
+};
+
 export const getPlatformAndVersion = (platform: string): string => {
   const platformShortName = getPlatformShortName(platform);
```

Comment:
you can move the call to `getPlatformShortName` inside the if block now, as it's used only there.
(be careful to add the brackets :D)